### PR TITLE
Fixes Nodropping items when arm is cut off

### PR DESCRIPTION
**Pull request by Gandalf2k15**
Fixes https://github.com/HippieStationCode/HippieStation13/issues/2515
Picture:
http://i.imgur.com/zDV2Kw7.png
Demo:
https://dl.dropboxusercontent.com/u/109295987/ShareX/2016/06/2016-06-20_23-36-01.mp4

### CODE REVIEW

**Comment by ghost** [comment]
So, what if someone chops your arm off and then picks up the no drop item?
What happens if you decap someone since with a hardsuit helmet?

**Comment by Gandalf2k15** [comment]
It stays on the suit, and is rehoused.. This fixes when your arm is cut off. When the other guy picks the suit up its just like it was. Its only a check for dismemberment.

File: code/modules/mob/inventory.dm [hunk 1]
```diff
@@ -93,17 +93,26 @@
 
 
 //Drops the item in our left hand
-/mob/proc/drop_l_hand() //I really fucking wonder why this proc had an argument holy shit.
-	if(!loc.allow_drop())
-		return
-	return unEquip(l_hand) //All needed checks are in unEquip
-
+/mob/proc/drop_l_hand(var/dismember) //I really fucking wonder why this proc had an argument holy shit.
+	if(dismember == 1)
```
Comment (JohnGinnane) [inline code/modules/mob/inventory.dm]: Couldn't this be shortened to:

```
if(!loc.allow_drop())
    return

if(dismember)
    unEquip(l_hand, 1)
else
    unEquip(l_hand)
```

Comment (Gandalf2k15) [inline code/modules/mob/inventory.dm]: Yes doing it

**Comment by JohnGinnane** [comment]
This explains why my fix didn't resolve this.. drop..hand wasn't allowing you to drop NODROP items

**Comment by JohnGinnane** [comment]
Code looks good, Gandalf I will merge when you do next commit and Travis is happy

**Comment by Gandalf2k15** [comment]
aye wait for travis

**Comment by Gandalf2k15** [comment]
fixed, wait for travis

**Comment by JohnGinnane** [comment]
Can I ask that you maybe add the argument into the proc calls here:

https://github.com/HippieStationCode/HippieStation13/blob/master/code/modules/mob/living/carbon/human/inventory.dm#L344

This way if for whatever reason the person does end up with an item in a non-existent hand it will automatically drop (even if NODROP)

File: code/modules/mob/inventory.dm [hunk 2]
```diff
@@ -93,17 +93,23 @@
...
 
 
 //Drops the item in our right hand
-/mob/proc/drop_r_hand()
+/mob/proc/drop_r_hand(var/dismember)
 	if(!loc.allow_drop())
 		return
-	return unEquip(r_hand) //Why was this not calling unEquip in the first place jesus fuck.
+	if(dismember)
+		unEquip(r_hand, 1)
```
Comment (JohnGinnane) [inline code/modules/mob/inventory.dm]: One last thing, i think we should return the output of these unEquips

Comment (Gandalf2k15) [inline code/modules/mob/inventory.dm]: Ah yes

**Comment by Gandalf2k15** [comment]
Alright fixed.

**Comment by JohnGinnane** [comment]
Your missing it in the drop_r_hand proc

**Comment by JohnGinnane** [comment]
Looks good. Merging this as it's a bug fix

**Pull request merged**